Rewrite Ubuntu install procedure

chenette opened the pull request:
This is a major rewrite/reorganization. Most of the text has changed. Also:

- I changed/added several commands, such as a step to start xo-tp-python
  (to be consistent with the Docker and AWS demo environments) and several
  commands to check the logs.

- I updated the output to match the current release.

Signed-off-by: Anne Chenette <chenette@bitwise.io>

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+      .. code-block:: console
 
-.. code-block:: console
+         [2018-03-14 16:00:17.223 INFO     processor_handlers] registered transaction processor: connection_id=eca3a9ad0ff1cdbc29e449cc61af4936bfcaf0e064952dd56615bc00bb9df64c4b01209d39ae062c555d3ddc5e3a9903f1a9e2d0fd2cdd47a9559ae3a78936ed, family=sawtooth_settings, version=1.0, namespaces=['000000']
```
I would only include this message

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
-which creates and submits a batch of transactions containing the
-settings change. This JSON array tells the validator or validator network
-to accept transactions of the following types:
+#. Start the XO transaction processor, ``xo-tp-python``. This transaction
```
I would make this an optional step, since it is only needed for the later tutorial.

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+  .. code-block:: console
 
-3. Submitting these transactions to the validator.
+     user@intkey-tp$ sudo -u sawtooth intkey-tp-python -v tcp://127.0.0.1:4004
```
This example is incorrect. It should be `--connect tcp://127.0.0.1:4004`. Also, the exact CLI arguments for transaction processors is implementation dependent and they are only consistent by convention, which should probably be mentioned. A link to the CLI reference would also be appropriate here.

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
 
-Viewing Blocks and State
-========================
+      user@client$ ps aux | grep [s]awtooth-rest-api
```
Why the `[s]awtooth` and not just `sawtooth`?

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+
+   .. code-block:: console
+
+      user@client$ sudo -u sawtooth sawtooth-rest-api -v
```
I would link back to the step for starting the REST API instead of repeating content from that section.

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+<../transaction_family_specifications/settings_transaction_family>`.
+
+#. Use the following ``sawset proposal create`` command to create and submit the
+   new settings to the validator (as a batch of transaction).
```
typo, "batch of transactions"

chenette replied:
Oops.  I missed that one.

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+
+      .. code-block:: console
+
+         [2018-03-14 16:24:49.587 DEBUG     received message of type: TP_PROCESS_REQUEST
```
This seems like an odd message to use as an indication that the batch was received. It might be better to direct the reader to the settings-tp terminal window instead, where they would see the transactions run, or to show the rest api submitting the batch to the validator with `[2018-03-15 14:21:38.058 DEBUG    route_handlers] Sending CLIENT_BATCH_SUBMIT_REQUEST request to validator`

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
 
+#. The validator terminal window displays many log messages showing that the
+   validator is handling the submitted transactions and processing blocks, as in
+   a this truncated example:
```
typo "a"

aludvik commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
+reason.
 
+#. Stop the validator by entering CTRL-c several times in the validator terminal
+   window.
```
see previous comment about stopping validator

chenette commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
-Transaction processors can be started either before or after the validator is
-started.
+#. Run the following command to start the REST API and connect to the local
+   validator.
```
I'll add a link for the REST API and `settings-tp`, but I can't do this for `intkey-tp-python` and `xo-tp-python` because they aren't in the CLI Command reference. Does anyone know why they're not there?

As a work-around (for now), I can say "Run ``intkey-tp-python --help`` and "Run ``xo-tp-python --help``".

chenette commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -2,488 +2,716 @@
 
-Creating and Submitting Transactions
-====================================
+Other Start-up Options for the Transaction Processors
```
To be consistent with the validator option change above, I'll delete this section.  I'll move the starting-order note to the top of the TP section. Note that the --bind info will move to the new section on adding nodes to an existing validator network.

chenette commented:
I have made the changes and corrections from Adam's review.  In addition to the simple corrections and wording tweaks, I made two significant changes: 

- I removed the two sections on other start-up options for the validator and transaction processors, and replaced them with a link to the relevant command in the CLI Command Reference.

- I kept the info about being able to start the transaction processors before the validator, but I reworded it and put it in the intro section as a note.
